### PR TITLE
Upgrade ScanCode to version 3.2.1rc2

### PR DESCRIPTION
**Pull request by fviernau**
Let's get the big upgrade done now and then make the minor migration from this pre-release to the final release later on.

### CODE REVIEW

**Review by tsteenbe — APPROVED** [review]
Approved, long overdue update...

**Review by sschuberth — DISMISSED** [review]

File: scanner/src/main/kotlin/scanners/ScanCode.kt [hunk 1]
```diff
@@ -197,7 +198,8 @@ class ScanCode(
         }
     }
 
-    override val scannerVersion = "3.0.2"
+    // Strip the 'rc2' suffix to avoid "SemverException: Invalid version (no patch version): 3.2.1rc2".
+    override val scannerVersion = SCANNER_VERSION.removeSuffix("rc2")
```
Comment (fviernau) [inline scanner/src/main/kotlin/scanners/ScanCode.kt:202]: I had no better idea to solve this. Anyone?

Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/ScanCode.kt:202]: Shouldn't inserting a dash before "rc2" work?

Comment (fviernau) [reply scanner/src/main/kotlin/scanners/ScanCode.kt:202]: No idea, need to investigate it.

Comment (fviernau) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: Applied an idea analog to yours.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**